### DISCUSSION TITLE
Creating vector embedding with LangChain and OpenAI

### DISCUSSION THREAD

**Discussion by 7matic** [milvus-io/milvus#26007] [Q&A and General discussion]
I just started using Milvus a few days ago so this will most likely be a naive question, but I would still appreciate if anyone could help me towards getting an answer.

I've been following this https://milvus.io/docs/integrate_with_langchain.md

Let's say I have a .pdf with data, I used LangChain to generate the embeddings and successfully saved everything inside just like it is shown in the link above. My question is whether I need to create these embeddings from given .pdf all the time so I could create question-answer system? It was shown how it is done for one time use, but not how could I do this over a longer period? So am I supposed to constantly do the steps that were provided there? It also takes about 15 seconds to get an answer so that's a huge limitation as well and I would like to improve that. I was wondering if it is possible that I can just do normal vector search after creating the vector embeddings from .pdf for the first time since the link provided above didn't discuss anything of that.

Thanks to everyone who is willing to help (and sorry for this basic question but I couldn't find a definitive answer regarding that). Have a nice day!

**Comment by yhmo** [comment]
"whether I need to create these embeddings from given .pdf all the time"-------- I think Yes. .pdf file is split into chunks and stored in Milvus. When you get a new .pdf file, you need to do the tokenizing/embedding process for it. Each .pdf file only does one time.
Milvus is a database, once all the .pdf files are successfully processed, you don't need to do the tokenizing/embedding again.
